### PR TITLE
106 overlay handle

### LINKED ISSUE #106: Signifier for Overlay Control
### Why/User Benefit/User Problem

Improve the user’s understanding of how the overlay works by adding a signifier to show it can be swiped up or down

### Acceptance Criteria (how do I know when I’m done?)

The homescreen card features a handle as a signifier and as an affordance that hints users that they can swipe up/down or drag up/down the homescreen card. This is possible also when the homescreen card features less than 4/8 tiles: [Mock](https://user-images.githubusercontent.com/1721875/46337872-ef46f400-c62e-11e8-8808-0c1c28587ca3.png)

### PR DESCRIPTION
**Pull request by Baron-Severin**
Closes #106 

### Pull Request checklist

- [X] This PR includes thorough **tests** or an explanation of why it does not
UI changes only
- [X] This PR includes a **CHANGELOG entry** or does not need one

### CODE REVIEW

File: app/src/main/res/layout/fragment_navigation_overlay.xml [hunk 1]
```diff
@@ -46,6 +46,17 @@
             app:layout_constraintStart_toStartOf="parent"
             android:background="@color/photonGrey70"/>
 
+    <View
```
Comment (mcomella) [inline app/src/main/res/layout/fragment_navigation_overlay.xml]: Add a comment that this view actually dismiss the overlay by allowing tap to fall through to the dismiss overlay view.

File: app/src/main/java/org/mozilla/focus/home/NavigationOverlayFragment.kt [hunk 2]
```diff
@@ -44,7 +44,7 @@ class NavigationOverlayFragment : Fragment() {
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         val overlay = inflater.inflate(R.layout.fragment_navigation_overlay, container, false)
 
-        val isVisibleInDialogMode = arrayOf(overlay.semiOpaqueBackground, overlay.dismissHitTarget)
+        val isVisibleInDialogMode = arrayOf(overlay.semiOpaqueBackground, overlay.dismissHitTarget, overlay.overlayHandle)
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/home/NavigationOverlayFragment.kt]: I'm so pleased with myself about how simple that was, lol.

File: app/src/main/java/org/mozilla/focus/widget/SwipeDownListener.kt [hunk 3]
```diff
@@ -10,6 +10,10 @@ import kotlin.math.abs
 
 /**
  * Executes [doOnSwipeDown] when down swipes are detected.
+ *
+ * TODO this was written to be used in #106, but related functionality was
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/widget/SwipeDownListener.kt]: I kind of want to remove this: I don't love confusing, unused code, especially if it accidentally sticks around for a few months, years. Maybe remove and squish those commits.

We can add it in another branch if you want to keep it around: maybe add it as a WIP comment in the other bug.

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [Unreleased]
 ### Added
 - Pinch to exit full screen web content like videos (#75)
+- "Handle" UI element above pinned tiles to help communicate swipe-to-dismiss functionality
```
Comment (mcomella) [inline CHANGELOG.md]: This is unnecessary: it's encapsulated in "Changed: appearance and animations of navigation overlay", which hasn't been released to users yet.

**Comment by mcomella** [comment]
Removing liuche as reviewer: I don't think it's necessary.

File: app/src/main/java/org/mozilla/focus/ext/MotionEvent.kt [hunk 5]
```diff
@@ -14,3 +16,12 @@ inline fun <R> MotionEvent.use(functionBlock: (MotionEvent) -> R): R {
         recycle()
     }
 }
+
+infix fun MotionEvent?.isWithinBoundsOf(view: View?): Boolean {
```
[the remark this answers is not in the document]
Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/focus/ext/MotionEvent.kt]: I think those are fair points, I'll change it.

**Comment by mcomella** [comment]
Let's leave this open for now: I think we'll want to land this eventually

**Comment by mcomella** [comment]
We can reopen this if it's still desired.

**Pull request closed without merging**